packages/identity/src/identity/ecdsa.ts

```diff
@@ -0,0 +1,349 @@
+import { DerEncodedPublicKey, PublicKey, Signature, SignIdentity } from '@dfinity/agent';
+import { SECP256K1_OID, unwrapDER, wrapDER } from './der';
+
+/**
+ * Options used in an {@link ECDSAPublicKey} or {@link ECDSAKeyIdentity}
+ */
+export type CryptoKeyOptions = {
+  extractable?: boolean;
+  keyUsages?: KeyUsage[];
+  subtleCrypto?: SubtleCrypto;
+};
+
+export class ExtractrableKeyError extends Error {
+  constructor(public readonly message: string) {
+    super(message);
+    Object.setPrototypeOf(this, ExtractrableKeyError.prototype);
+  }
+}
+
+export class CryptoError extends Error {
+  constructor(public readonly message: string) {
+    super(message);
+    Object.setPrototypeOf(this, ExtractrableKeyError.prototype);
+  }
+}
+
+/**
+ * Utility method to ensure that a subtleCrypto implementation is provided or is available in the global context
+ * @param subtleCrypto SubtleCrypto implementation
+ * @returns subleCrypto
+ */
+function _getEffectiveCrypto(subtleCrypto: CryptoKeyOptions['subtleCrypto']): SubtleCrypto {
+  if (subtleCrypto) {
+    return subtleCrypto;
+  } else if (typeof crypto !== 'undefined' && crypto['subtle']) {
+    return crypto.subtle;
+  } else {
+    throw new CryptoError(
+      'Global crypto was not available and none was provided. Please inlcude a SubtleCrypto implementation. See https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto',
+    );
+  }
+}
+
+/**
+ * A public key interface that wraps an ECDSA key using the P-256 named curve. Supports DER-encoding and decoding for agent calls
+ */
+export class ECDSAPublicKey implements PublicKey {
+  private static derEncode(publicKey: ArrayBuffer): DerEncodedPublicKey {
+    // TODO - replace placeholder DER logic
```

Comment:
What does this mean?  Is this something yet to be done in a PR, or should we track this?

Reply:
That was resolved, if I understand correctly that a ECDSA P-256 signature should use the same DER encoding as secp256k1. CC @Dfinity-Bjoern @andrew-lee-work

Reply:
Fixed using `spki` export strategy